### PR TITLE
Adding mode flag to train command

### LINKED ISSUE #1087: Add a flag to trigger alternate training mode to BF LUIS:Train
LUIS vNext is working on new training technology. In order to offer the capability to selected customers as well as create a preview offering by BUILD the LUIS team requests that we add a new flag to LUIS:train command. The requested capability should be available at the underlying library so it is accessible by Composer.

The only additional query parameter needs to be done for the Train API to use the private preview capabilities, and it is optional.
 
New query parameter name is mode. Default value is Standard (corresponding to regular old training) and private preview training value is Neural (corresponding to new deep stack training)
 
          {
            "in": "query",
            "name": "mode",
            "description": "An enum value specifying mode of training.",
            "type": "string",
            "enum": [
              "Standard",
              "Neural"
            ]
          }
 
If the request does not contain the enum, it will fall back to Standard.

* If we sent the feature flag with mode set to ‘neural’, but the customer is not in the private preview, will the service fall back to standard?
  - Correct
* How long before you plan to broaden the preview beyond a very small number of customers?
  - The tentative plan is //BUILD but this depends on the feedback that we receive from our private preview.
* Have you already identified the limited set of private preview customers? If so, it would be good to align and understand which of them are using Composer today.
  - We chatted with Darren yesterday and I am looking up to help from your side to recommend existing customers or new leads that would be candidates for our private preview

### PR DESCRIPTION
**Pull request by munozemilio**
Fixes #1087

### CODE REVIEW

File: packages/luis/src/api/http-request.ts [hunk 1]
```diff
@@ -78,7 +78,12 @@ const httpRequest = async function (subscriptionKey: string, config: any) {
     if (error.response) {
       // The request was made and the server responded with a status code
       // that falls out of the range of 2xx
-      throw Error(error.response.statusText)
+      let message = ''
```
Comment (joshgummersall) [inline packages/luis/src/api/http-request.ts:81]: As long as the Typescript version is current, `const message = error?.response?.data?.error?.message ?? ''` should cover all the checks.

File: packages/luis/src/commands/luis/train/run.ts [hunk 2]
```diff
@@ -37,7 +38,7 @@ export default class LuisTrainRun extends Command {
     utils.validateRequiredProps(requiredProps)
 
     try {
-      const trainingRequestStatus = await Train.train({subscriptionKey, endpoint, appId}, versionId)
+      const trainingRequestStatus = await Train.train({subscriptionKey, endpoint, appId}, versionId, flags.mode)
```
Comment (joshgummersall) [inline packages/luis/src/commands/luis/train/run.ts:41]: Are string flags defaulted to `''`?

**Review by joshgummersall — APPROVED** [review]
One suggestion, one question. LGTM though.

File: packages/luis/src/commands/luis/train/run.ts (hunk 2 above)
Comment (munozemilio) [reply packages/luis/src/commands/luis/train/run.ts:41]: to undefined

Comment (joshgummersall) [reply packages/luis/src/commands/luis/train/run.ts:41]: It's pedantic, but if `undefined` is a valid type for the `mode` parameter, the `train` function signature should be `(..., mode?: string)`. `strict` mode would raise an error here.

**Pull request merged**